packages/components/src/font-size-picker/index.js

```diff
@@ -117,33 +131,20 @@ function FontSizePicker(
 					/>
 				) }
 				{ ! withSlider && ! disableCustomFontSizes && (
-					<div className="components-font-size-picker__number-container">
-						<label htmlFor={ fontSizePickerNumberId }>
-							{ __( 'Custom' ) }
-						</label>
-						<input
-							id={ fontSizePickerNumberId }
-							className="components-font-size-picker__number"
-							type="number"
-							min={ 1 }
-							onChange={ ( event ) => {
-								if (
-									! event.target.value &&
-									event.target.value !== 0
-								) {
-									onChange( undefined );
-									return;
-								}
-								if ( hasUnits ) {
-									onChange( event.target.value + 'px' );
-								} else {
-									onChange( Number( event.target.value ) );
-								}
-							} }
-							aria-label={ __( 'Custom' ) }
-							value={ ( isPixelValue && noUnitsValue ) || '' }
-						/>
-					</div>
+					<UnitControl
+						label={ __( 'Custom' ) }
+						labelPosition="top"
+						__unstableInputWidth="60px"
+						value={ value }
+						onChange={ ( nextSize ) => {
+							if ( 0 === parseFloat( nextSize ) || ! nextSize ) {
+								onChange( undefined );
+							} else {
+								onChange( nextSize );
+							}
+						} }
+						units={ CSS_UNITS }
```

Comment:
I think themes can control the available units or disable them, is this being taken into consideration here?